lib/metadata/jsonMetadata.hpp

```diff
@@ -0,0 +1,151 @@
+#ifndef JSON_METADATA
+#define JSON_METADATA
+
+
+#include "json.hpp"
+
+#include <cstdint>
+#include <string>
+#include <ctime>
+#include <vector>
+
+#include <sys/time.h>
+
+#pragma pack()
+
+// json based metadata structures to hold "physics" metadata, ie. data that
+// various at most once per frame
+
+namespace jsonMetadata {
+// these must match chimeMetadata and choordMetadata for now
+// in namespace for C++ const
+#define MAX_NUM_BEAMS 20
```

Comment:
This max beams logic was largely in place to deal with the fact we used structs which had compile time defined size.   We can remove that now that we have a dynamic structure.